lib/cache/collections.go

```diff
@@ -137,6 +137,11 @@ func setupCollections(c Config) (*collections, error) {
 		resourceKind := resourceKindFromWatchKind(watch)
 
 		switch watch.Kind {
+		case types.KindAccessRequest, types.KindHeadlessAuthentication:
+			// These types are not stored in the cache itself, they are only
+			// ever cached so that the resources events can be processed by
+			// downstream watchers.
+			continue
```

Comment:
What do you think of making this a separate function?

```suggestion
		if isKnownUncollectedKind(watch.Kind) {
			continue
		}
		switch watch.Kind {
```

```go
// isKnownUncollectedKind is true if a resource kind is not stored in
// the cache itself but it's only configured in the cache so that the
// resources events can be processed by downstream watchers.
func isKnownUncollectedKind(kind string) bool {
	switch kind {
	case types.KindAccessRequest, types.KindHeadlessAuthentication:
		return true
	default:
		return false
	}
}
```